### PR TITLE
Add support for cross-compilation, with backcompat

### PR DESCRIPTION
**Pull request by Ericson2314**
In https://github.com/kolloch/crate2nix/pull/152#issuecomment-754203498 I made a PR against a PR, but i get not CI that way, so I'm opening this too.

### CODE REVIEW

File: crate2nix/templates/nix/crate2nix/default.nix [hunk 1]
```diff
@@ -5,6 +5,9 @@
 , lib
 , stdenv
 , buildRustCrate
+, buildRustCrateForPkgs ? if buildRustCrate != null
+  then lib.warn "`buildRustCrate` is deprecated, use `buildRustCrateForPkgs` instead" (_: buildRustCrate)
```
Comment (kolloch) [inline crate2nix/templates/nix/crate2nix/default.nix:9]: I really dig this user guidance, thank you 👍

File: crate2nix/templates/Cargo.nix.tera [hunk 2]
```diff
@@ -9,7 +9,11 @@
 , pkgs ? import nixpkgs { config = {}; }
 , lib ? pkgs.lib
 , stdenv ? pkgs.stdenv
-, buildRustCrate ? pkgs.buildRustCrate
+, buildRustCrateForPkgs ? if buildRustCrate != null
+    then lib.warn "`buildRustCrate` is deprecated, use `buildRustCrateForPkgs` instead" (_: buildRustCrate)
+    else pkgs: pkgs.buildRustCrate
+  # Deprecated
+, buildRustCrate ? null
   # This is used as the `crateOverrides` argument for `buildRustCrate`.
```
Comment (kolloch) [inline crate2nix/templates/Cargo.nix.tera:17]: Update comment?

**Review by kolloch — APPROVED** [review]
Let's merge this :)

I'd super appreciate it if you contributed some guidance on how to use that for the README.md.

File: crate2nix/templates/Cargo.nix.tera (hunk 2 above)
Comment (Ericson2314) [reply crate2nix/templates/Cargo.nix.tera:17]: Oh! perhaps I was confused whether the comments were describing the thing above or the thing below.

**Pull request merged**